docs/en/concepts-and-designs/lal.md

```diff
@@ -240,6 +240,80 @@ metrics:
     exp: http_response_time.sum(['le', 'service', 'instance']).increase('PT5M').histogram().histogram_percentile([50,70,90,99])
 ```
 
+### SlowSql
+
+SlowSql aims to convert LogData to DatabaseSlowStatement. It extracts data from `parsed` result and save them as DatabaseSlowStatement. SlowSql will not abort or edit logs, you can use other LAL for further processing.
+We need to set a log tag("isSlowSql") to enable it. If logs sent to OAP does not have this tag, slowSql processing will be skipped.The example to set the tag is following:
+``` json
+[
+   {
+      "tags":{
+         "data":[
+            {
+               "key":"isSlowSql",
+               "value":"true"
+            }
+         ]
+      },
+      "body":{
+         "json":{
+            "json":"{\"time\":\"20220906153959\",\"id\":\"cb92c1a5b-2691e-fb2f-457a-9c72a392d9ed\",\"service\":\"root[root]@[localhost]\",\"statement\":\"select sleep(2);\",\"layer\":\"MYSQL\",\"query_time\":2000}"
+         }
+      },
+      "service":"root[root]@[localhost]"
+   }
+]
+```
+
+- `serviceName`
+
+`serviceName` extracts the service name from the `parsed` result, and set it into the `DatabaseSlowStatement`, which will be persisted (if
+not dropped) and is used to associate with TopNDatabaseStatement and service.
+
+- `timeBucket`
+
+`timeBucket` extracts the time bucket from the `parsed` result, and set it into the `DatabaseSlowStatement`, which will be
+persisted (if not dropped) and is used to associate with TopNDatabaseStatement.
+
+- `statement`
+
+`statement` extracts the time bucket from the `parsed` result, and set it into the `DatabaseSlowStatement`, which will be
+persisted (if not dropped) and is used to associate with TopNDatabaseStatement.
+
+- `latency`
+
+`latency` extracts the latency from the `parsed` result, and set it into the `DatabaseSlowStatement`, which will be
+persisted (if not dropped) and is used to associate with TopNDatabaseStatement.
+
+- `id`
+
+`id` extracts the id from the `parsed` result, and set it into the `DatabaseSlowStatement`, which will be persisted (if not
+dropped) and is used to associate with TopNDatabaseStatement.
+
+- `layer`
+
+`layer` extracts the [layer](../../../oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/Layer.java) from the `parsed` result, and set it into the `DatabaseSlowStatement`, which will be persisted (if
+not dropped) and is used to associate with service.
+
+Example :
+
+```groovy
+filter {
+        json{
+        }
+        if (log.tags.getDataList().find{tag1->tag1.getKey() == "isSlowSql"}.getValue() == "true") {
```

Comment:
This can be a trouble for the users who don't understand the Groovy syntax, let's simplify this, can you add a convenient getter method `String tag(String key)` in `ExtractorSpec` class, and change this example to:


```groovy
filter {
  json {
  }
  if (log.tag("isSlowSql") == "true") {
  }
}
```

Reply:
I understand.

Reply:
@kezhenxu94 Modifying Class `ExtractorSpec` may not have this effect, maybe I need to modify the 
Class `LALPrecompiledExtension`?

Reply:
> @kezhenxu94 Modifying Class `ExtractorSpec` may not have this effect, maybe I need to modify the
> Class `LALPrecompiledExtension` and `AbstractSpec`?

Hi, feel free to modify those if needed, I just provided an idea and it's not complete.

Reply:
I have added a global method to resolve it.
```
filter {
        json{
        }
        extractor{
          layer parsed.layer as String
          service parsed.service as String
          timestamp parsed.time as String
          if (tag("LOG_KIND") == "SLOW_SQL") {
             slowSql {
                      id parsed.id as String
                      statement parsed.statement as String
                      latency parsed.query_time as Long
                     }
          }
        }
      }
```